apps/namada-interface/src/slices/fees.ts

```diff
@@ -0,0 +1,66 @@
+import { Query } from "@namada/shared";
+import BigNumber from "bignumber.js";
+import { atom } from "jotai";
+
+import { accountsAtom } from "slices/accounts";
+import { chainAtom } from "slices/chain";
+
+// TODO: remove harcoding of gas limit
+export const GAS_LIMIT = new BigNumber(20_000);
+
+const minimumGasPriceAtom = (() => {
+  const base = atom(new Promise<BigNumber>(() => {}));
+
+  return atom(
+    (get) => get(base),
+    async (get, set) => {
+      const {
+        rpc,
+        currency: { address: nativeToken },
+      } = get(chainAtom);
+      const query = new Query(rpc);
+
+      const promise = (async () => {
+        const result = (await query.query_gas_costs()) as [string, string][];
+        const nativeTokenCost = result.find(([token]) => token === nativeToken);
+
+        if (!nativeTokenCost) {
+          throw new Error("Error querying minimum gas price");
+        }
+
+        const asBigNumber = new BigNumber(nativeTokenCost[1]);
+
+        if (asBigNumber.isNaN()) {
+          throw new Error("Error converting minimum gas price to BigNumber");
+        }
+
+        return asBigNumber;
+      })();
+
+      set(base, promise);
+    }
+  );
+})();
+
+const isRevealPkNeededAtom = atom(async (get) => {
+  const accounts = await get(accountsAtom);
+  const transparentAccounts = accounts.filter((account) => !account.isShielded);
+
+  const { rpc } = get(chainAtom);
+  const query = new Query(rpc);
+
+  const map: Record<string, string | null> = {};
+  transparentAccounts.forEach(async ({ address }) => {
```

Comment:
Also if we want to do async/await in loop we need to use `for...of` syntax
```ts
  for await (const { address } of transparentAccounts) { }
```
I think this works because the returned function is not being called immediately

Reply:
Is that the case only if we want each call to the callback to await the previous one finishing?
In this case, I want each call to the callback to run in parallel, but within the callback I need to await the public key in order to use its value.
Basically I'm imagining if it was `map` instead of `forEach`, the return value would be an array of promises which are independent of each other, but since it's `forEach` those promises get thrown away. Have I got this reasoning wrong somewhere?

Reply:
If that's the case then this is fine. It might be a bit confusing because when you want to read the map immediately after the forEach, it's value is `{}` as forEach just calls async functions in loop :P Not sure if we need all of the promises to be resolved, but maybe Promise.all would be a good solution here? :D But yeah if we do not want to wait then this is fine

Reply:
Ah wait, yeah you're totally right there that I should use `Promise.all`, because I was returning the function before the queries have finished. Maybe I need to think about the fetching behaviour a bit more in a later PR, but for now I will just await all public key queries before returning. Thanks for the explanation!